### PR TITLE
Add `esc-to-cancel` feature

### LINKED ISSUE #3692: Add a keyboard shortcut to cancel editing title
Add a keyboard shortcut to cancel editing title (e.g. <kbd>esc</kbd>, just like #961 when editing a comment).

### PR DESCRIPTION
**Pull request by Liberatys**
1. LINKED ISSUES:
   Closes #3692

2. TEST URLS:
   https://github.com/sindresorhus/refined-github/pull/3701

3. SCREENSHOT:
   
Screen Recording 2020-11-05 at 23 20 24

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -215,6 +215,7 @@ Thanks for contributing! 🦋🙌
 - [](# "warn-pr-from-master") [Warns you when creating a pull request from the default branch, as it’s an anti-pattern.](https://user-images.githubusercontent.com/1402241/52543516-3ca94e00-2de5-11e9-9f80-ff8f9fe8bdc4.png)
 - [](# "warning-for-disallow-edits") [Warns you when unchecking `Allow edits from maintainers`, as it’s maintainer-hostile.](https://user-images.githubusercontent.com/1402241/53151888-24101380-35ef-11e9-8d30-d6315ad97325.gif)
 - [](# "pull-request-hotkey") [Adds keyboard shortcuts to cycle through PR tabs: <kbd>g</kbd> <kbd>←</kbd> and <kbd>g</kbd> <kbd>→</kbd>, or <kbd>g</kbd> <kbd>1</kbd>, <kbd>g</kbd> <kbd>2</kbd>, <kbd>g</kbd> <kbd>3</kbd> and <kbd>g</kbd> <kbd>4</kbd>](https://user-images.githubusercontent.com/16872793/94634958-7e7b5680-029f-11eb-82ea-1f96cd11e4cd.png).
+- [](# "esc-to-cancel") [Adds a shortcut to stop editing the issue title: <kbd>esc</kbd>.](https://github.com/sindresorhus/refined-github/pull/3701)
```
Comment (Liberatys) [inline readme.md:218]: Not sure about the wording .. maybe cancel edit is better

File: source/github-events/on-issue-title-field-keydown.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import select from 'select-dom';
+import delegate from 'delegate-it';
+
+export default function onIssueTitleKeydown(callback: delegate.EventHandler<KeyboardEvent, HTMLTextAreaElement>): void {
```
Comment (fregante) [inline source/github-events/on-issue-title-field-keydown.tsx:4]: This doesn’t need to be its own function/file. I don’t think there’s any autocomplete on the title.

**Comment by fregante** [comment]
Thanks for the PR, Nick!

File: source/features/esc-to-cancel.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import onIssueTitleKeydown from '../github-events/on-issue-title-field-keydown';
+
+function eventHandler(event: delegate.Event<KeyboardEvent, HTMLTextAreaElement>): void {
+	const field = event.delegateTarget;
+	if (event.key === 'Escape') {
+		const cancelButton = select<HTMLLinkElement>(`
+				.js-cancel-issue-edit
+			`, field.form!);
+		// Cancel if there is a button, else blur the field
+		if (cancelButton) {
```
Comment (fregante) [inline source/features/esc-to-cancel.tsx:15]: For example `cancel` should **always** exist here, so no need for this condition

File: readme.md [hunk 4]
```diff
@@ -215,6 +215,7 @@ Thanks for contributing! 🦋🙌
 - [](# "warn-pr-from-master") [Warns you when creating a pull request from the default branch, as it’s an anti-pattern.](https://user-images.githubusercontent.com/1402241/52543516-3ca94e00-2de5-11e9-9f80-ff8f9fe8bdc4.png)
 - [](# "warning-for-disallow-edits") [Warns you when unchecking `Allow edits from maintainers`, as it’s maintainer-hostile.](https://user-images.githubusercontent.com/1402241/53151888-24101380-35ef-11e9-8d30-d6315ad97325.gif)
 - [](# "pull-request-hotkey") [Adds keyboard shortcuts to cycle through PR tabs: <kbd>g</kbd> <kbd>←</kbd> and <kbd>g</kbd> <kbd>→</kbd>, or <kbd>g</kbd> <kbd>1</kbd>, <kbd>g</kbd> <kbd>2</kbd>, <kbd>g</kbd> <kbd>3</kbd> and <kbd>g</kbd> <kbd>4</kbd>](https://user-images.githubusercontent.com/16872793/94634958-7e7b5680-029f-11eb-82ea-1f96cd11e4cd.png).
+- [](# "esc-to-cancel") [Adds a shortcut to cancel issue title editing: <kbd>esc</kbd>.](https://github.com/sindresorhus/refined-github/pull/3701)
```
Comment (fregante) [inline readme.md:218]: This must be a screenshot or gif, not a link to a PR

**Review by fregante — APPROVED** [review]

File: readme.md [hunk 5]
```diff
@@ -215,6 +215,7 @@ Thanks for contributing! 🦋🙌
 - [](# "warn-pr-from-master") [Warns you when creating a pull request from the default branch, as it’s an anti-pattern.](https://user-images.githubusercontent.com/1402241/52543516-3ca94e00-2de5-11e9-9f80-ff8f9fe8bdc4.png)
 - [](# "warning-for-disallow-edits") [Warns you when unchecking `Allow edits from maintainers`, as it’s maintainer-hostile.](https://user-images.githubusercontent.com/1402241/53151888-24101380-35ef-11e9-8d30-d6315ad97325.gif)
 - [](# "pull-request-hotkey") [Adds keyboard shortcuts to cycle through PR tabs: <kbd>g</kbd> <kbd>←</kbd> and <kbd>g</kbd> <kbd>→</kbd>, or <kbd>g</kbd> <kbd>1</kbd>, <kbd>g</kbd> <kbd>2</kbd>, <kbd>g</kbd> <kbd>3</kbd> and <kbd>g</kbd> <kbd>4</kbd>](https://user-images.githubusercontent.com/16872793/94634958-7e7b5680-029f-11eb-82ea-1f96cd11e4cd.png).
+- [](# "esc-to-cancel") [Adds a shortcut to cancel issue title editing: <kbd>esc</kbd>.](https://user-images.githubusercontent.com/35100156/98303086-d81d2200-1fbd-11eb-8529-70d48d889bcf.gif)
```
Comment (yakov116) [inline readme.md:218]: Shouldn't this go on the bottom of the list?

**Review by yakov116 — APPROVED** [review]
Just a small comment.

Great job!

**Pull request merged**